commands/migratecmd.go

```diff
@@ -264,6 +267,11 @@ func (mr *MigrateRunner) migrateObjs(cmObjs []object.ObjMetadata, reader io.Read
 	if err != nil {
 		return err
 	}
+	err = updateOwningInventoryAnnotation(mr.rgProvider.Factory(), cmObjs, oldID, inv.ID())
```

Comment:
So in this case we add the `owning-inventory` annotation regardless of whether the resource had one already right?
If something should fail during migration and kpt crashes just before this line, would re-running the migration still result in the correct end state? I think so, but I just want to make sure we have thought that through.

Reply:
This function checks if the resource has the `owning-inventory` annotation. If it doesn't have or if the `owning-inventory` points to the previous ConfigMap inventory object, it updates the `owning-inventory` to the new ResourceGroup object. The reason of doing this is that the `owning-inventory` might be other inventory object and we can't always change it to the ResourceGroup object during migration. If the `owning-inventory` is already the ResourceGroup object, we don't need to update it; if the `owning-inventory` is another object, the migration can't adopt it to the current ResourceGroup object. 

Since the current logic is to create the ResourceGroup CR first, then update the `owning-inventory` annotation. In the case you mentioned, if the ResourceGroup object is created, re-running the migration would skip the part for adding `owning-inventory` annotation.  If the ResourceGroup object wasn't created, re-running the migration would result in the correct end state.

If we change it to first update the `owning-inventory` annotations; then create the ResourceGroup CR. If there is any failure during the migration, re-running the migration would fix it.

Reply:
Changed the order of the two operations. PTAL.